src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -49,11 +49,12 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
             IComparer<Transaction> comparer = GetComparer(parent, new BlockPreparationContext(baseFee, blockNumber))
                 .ThenBy(ByHashTxComparer.Instance); // in order to sort properly and not lose transactions we need to differentiate on their identity which provided comparer might not be doing
 
-            Func<Transaction, bool> filter = (tx) => _txFilterPipeline.Execute(tx, parent);
 
-            IEnumerable<Transaction> transactions = GetOrderedTransactions(pendingTransactions, comparer, filter, gasLimit);
+            Func<Transaction, bool> filter = (tx) => _txFilterPipeline.Execute(tx, parent, spec);
+
+            IEnumerable<Transaction> transactions = GetOrderedTransactions(pendingTransactions, comparer, filter, gasLimit).ToArray();
```

Comment:
we explicitly tried to avoid `ToArray` here

Reply:
Yep sure, it was added just for testing